File: python/upload_dynamodb.py
import boto3
from data_dump import parse_xml_file, get_properties, filter_properties_by_city
import time

dynamodb = boto3.resource('dynamodb')
table_name = 'rentable_properties'

table = dynamodb.Table(table_name)

def add_properties_to_table():
    file_path = 'python/rentable_input_data.xml'
    root = parse_xml_file(file_path)
    properties = get_properties(root)
    print("Total properties:", len(properties))
    filtered_properties = filter_properties_by_city(properties, 'Madison')
    print("Total properties in city:", len(filtered_properties))
    start_time = time.time()

    for prop in filtered_properties:
        table.put_item(Item=prop)

    end_time = time.time()
    run_time = end_time - start_time
    print(f'Run completed in {run_time:.2f} seconds')

add_properties_to_table()

File: ruby/upload_dynamodb.rb
require 'aws-sdk-dynamodb'
require 'nokogiri'
require_relative 'data_dump'
require 'pry'

def add_properties_to_table()
    dynamodb = Aws::DynamoDB::Resource.new(region: 'us-east-1')
    table_name = 'properties'
    table = dynamodb.table(table_name)
    file_path = 'ruby/rentable_input_data.xml'
    doc = parse_xml_file(file_path)
    properties = get_properties(doc)
    filtered_properties = filter_properties_by_city(properties, 'Madison')
    
    filtered_properties.each do |prop|
        table.put_item({ item: prop })
    end
end

add_properties_to_table
